csrc/abstract_tensor.h

```diff
@@ -127,6 +127,25 @@ using AbstractId = dynamic_type::DynamicType<
 //   AbstractTensor v({{id0, id1}, id2});
 //   auto ub = v.unzip();
 // Then ub will be {AbstractTensor{id0, id2}, AbstractTensor{id1, id2}}
+//
+// The opposite operation of unzip is zip. For example:
+//
+// Example 10:
+//   IterDomain *id0, *id1, *id2, *id3;
+//   AbstractTensor v0({id0, id2});
+//   AbstractTensor v1({id1, id3});
+//   auto z = AbstractTensor::zip({v0, v1});
+// Then z will be [{id0, id1}, {id2, id3}].
+//
+// Besides, you can stack another AbstractTensor into the current
+// AbstractTensor. For example:
+//
+// Example 11:
+//   IterDomain *id0, *id1, *id2, *id3, *id4, *id5;
+//   AbstractTensor v0({{id0, id1}, {id2, id3}});
+//   AbstractTensor v1({id4, id5});
+//   v0.stack(v1);
+// Then v0 will be [{id0, id1, id4}, {id2, id3, id5}].
```

Comment:
Is `zip` just a special case of `stack` where all the input `AbstractId`s are non-vectors? I wonder if we could just implement a general stack that handles both vector and non-vector `AbstractId`s in each position and then alias it as `zip` for ergonomics.

Reply:
Good question. I think the primary concern I have is, "handles both vector and non-vector" is not very well defined. For example, if I have two abstract tensors:
```
t0 = {id0, {id1, id2}}
t1 = {id3, id4}
```
then what should the stack of `t0` with `t1` defined? Should the result be

```
{{id0, id3}, {id1, id2, id4}}
```

or 

```
{{id0, id3}, {{id1, id2}, id4}}
```